android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/UrlDeviceDiscoveryService.java

```diff
@@ -183,17 +185,26 @@ private void restoreCache() {
       JSONObject serializedCollection = new JSONObject(prefs.getString(PW_COLLECTION_KEY, null));
       mPwCollection = PhysicalWebCollection.jsonDeserialize(serializedCollection);
       Utils.setPwsEndpoint(this, mPwCollection);
-      // replace TxPower and RSSI data after restoring cache
-      for (UrlDevice urlDevice : mPwCollection.getUrlDevices()) {
-        if (Utils.isBleUrlDevice(urlDevice)) {
-          Utils.updateRegion(urlDevice);
-        }
-      }
     } catch (JSONException e) {
       Log.e(TAG, "Could not restore Physical Web collection cache", e);
     } catch (PhysicalWebCollectionException e) {
       Log.e(TAG, "Could not restore Physical Web collection cache", e);
     }
+    // replace TxPower and RSSI data after restoring cache
+    for (UrlDevice urlDevice : mPwCollection.getUrlDevices()) {
+      if (Utils.isBleUrlDevice(urlDevice)) {
+        Utils.updateRegion(urlDevice);
+      }
+    }
+    // if the unresolvable devices cache is
```

Comment:
Ideally, use complete sentences.  Say something like "Unresolvable devices are typically not relevant outside of scan range.  Hence, we specially clean them from the cache."